client/wayland/ibuswaylandim.c

```diff
@@ -1633,13 +2103,30 @@ ibus_wayland_im_set_surface (IBusWaylandIM *wlim,
     g_return_val_if_fail (wlim, FALSE);
     g_return_val_if_fail (_surface, FALSE);
     priv = ibus_wayland_im_get_instance_private (wlim);
-    if (!priv->panel) {
-        g_warning ("Need zwp_input_panel_v1 before the surface setting.");
-        return FALSE;
+    switch (priv->version) {
```

Comment:
I have searched the code but I couldn't figure out where `ibus_wayland_im_set_surface` is actually called. Shouldn't this one be hooked up somewhere still?

Reply:
It's called by the panel instance as `IBus.WaylandIM.set_surface()`:
https://github.com/ibus/ibus/blob/main/ui/gtk3/application.vala#L287

Reply:
ah, thanks for the hint! My search must have missed it for some reason